Allow synthflesh removal from pustule with syringe

Saicchi opened the pull request:
<!-- The text between the arrows are comments - they won't be visible on your PR. -->
<!-- To automatically tag this PR, add the label(s) surrounded by brackets anywhere, for example: [LABEL] -->
<!-- PRs should at least have one area (A-) label and at least one category (C-) label -->

[ENHANCEMENT][CHEMISTRY]

## About the PR <!-- Describe the Pull Request here. What does it change? What other things could this impact? -->

Allows removal of synthflesh from pustules using syringes using a 1 second interruptible action. Process is only one way.

A few seconds after poking, the pustule will attempt to heal it using 15 units of synthflesh and reducing efficiency by 20% to prevent it from being abused. If it doesn't have enough to fix the hole it gets destroyed instead.

## Why's this needed? <!-- Describe why you think this should be added to the game. -->

Cyborgs can't make synthflesh. It requires picking it up and plopping them on a beaker or barrel, which they can't do.

## Changelog
<!-- If necessary, put your changelog entry below. Otherwise, please delete this section.
Use however you want to be credited in the changelog in place of CodeDude.
Use (*) for major changes and (+) for minor changes. For example: -->

```changelog
(u)DrWolfy
(+)Synthflesh in pustules can now be recovered with a syringe, reducing the pustule efficiency.
```

Azrun commented on code/modules/chemistry/tools/syringes.dm:
```diff
@@ -116,6 +116,16 @@
+				if (istype(target,/obj/item/reagent_containers/synthflesh_pustule))
+					var/obj/item/reagent_containers/synthflesh_pustule/P = target
+					P.reagents.trans_to(src, src.amount_per_transfer_from_this)
+					logTheThing(LOG_CHEMISTRY, user, "draws 5 units of reagents from [constructTarget(target,"combat")] [log_reagents(target)] with a syringe [log_reagents(src)] at [log_loc(user)].")
```
5 ->                             src.amount_per_transfer_from_this?

Saicchi replied:
Copied from a few lines below, will fix that one too.

Azrun commented on code/modules/chemistry/tools/syringes.dm:
```diff
@@ -116,6 +116,16 @@
+					var/obj/item/reagent_containers/synthflesh_pustule/P = target
+					P.reagents.trans_to(src, src.amount_per_transfer_from_this)
+					logTheThing(LOG_CHEMISTRY, user, "draws 5 units of reagents from [constructTarget(target,"combat")] [log_reagents(target)] with a syringe [log_reagents(src)] at [log_loc(user)].")
+					if (!P.poked)
+						P.poked = TRUE
+						SPAWN(3 SECONDS)
+							P.try_healpoke()
```
Is the intent you have 3 seconds to use the hole?

Saicchi replied:
Yes. Giving the player 3 seconds until it closes and they need to poke a new one.

Azrun commented on code/modules/chemistry/tools/syringes.dm:
```diff
@@ -116,6 +116,16 @@
+					logTheThing(LOG_CHEMISTRY, user, "draws 5 units of reagents from [constructTarget(target,"combat")] [log_reagents(target)] with a syringe [log_reagents(src)] at [log_loc(user)].")
+					if (!P.poked)
+						P.poked = TRUE
+						SPAWN(3 SECONDS)
```
Why not use the `process` loop?

Saicchi replied:
Removed the SPAWN and moved it to a process loop.

Saicchi commented:
Updated the PR.
Drawing from the pustule using a syringe is now a 1 second interruptible action like a player instead of being instant.
Moved the loop to check if it was poked to a `process` loop.

Flaborized commented:
As the person who made the pustules, I'm not against borgs being able to utilize them and make synthflesh, but I think syringes are a very tedious way to do that and I wanted the 'despositing synthflesh from the pustule' part of the process to be quick and painless. I'd much rather it be doable in some comparably instant way to how humans can do it

Saicchi commented:
> As the person who made the pustules, I'm not against borgs being able to utilize them and make synthflesh, but I think syringes are a very tedious way to do that and I wanted the 'despositing synthflesh from the pustule' part of the process to be quick and painless. I'd much rather it be doable in some comparably instant way to how humans can do it

I did think about letting borgs just click and drag the pustule to the barrel, and I much prefer that option, but midway through it I gave up since it seemed a bit imbalanced as they don't need to pick up the pustule and risk their arm getting ripped without calming it down first.

frawhst commented on code/modules/chemistry/Chemistry-Tools.dm:
```diff
@@ -911,6 +911,9 @@ proc/ui_describe_reagents(atom/A)
 	var/list/convertible_reagents = list("blood","bloodc","bloody_mary","bloody_scary","hemolymph")
+	var/poked = FALSE // poked by a syringe, will try to heal using synthflesh
+	var/pokes = 0 // each poke reduces efficiency
+	var/poke_timer = 2
```
Would appreciate some comments on how `poke_timer` works, I think I get it, but it's done in imprecise Per Process Tick units so it's hard to understand from the variable definition alone

frawhst commented on code/modules/chemistry/Chemistry-Tools.dm:
```diff
@@ -911,6 +911,9 @@ proc/ui_describe_reagents(atom/A)
 	var/angry = FALSE
 	var/angry_timer = 15
 	var/list/convertible_reagents = list("blood","bloodc","bloody_mary","bloody_scary","hemolymph")
+	var/poked = FALSE // poked by a syringe, will try to heal using synthflesh
+	var/pokes = 0 // each poke reduces efficiency
```
```suggestion
	var/poked = FALSE //! poked by a syringe, will try to heal using synthflesh
	var/pokes = 0 //! each poke reduces efficiency
```
code doc comments, can be read on hover and are generally good

frawhst commented on code/modules/chemistry/Chemistry-Tools.dm:
```diff
@@ -923,6 +926,15 @@ proc/ui_describe_reagents(atom/A)
+	examine(mob/user)
+		. = ..()
+		if (pokes > 5)
+			. += "Its full of patched up holes!"
```
```suggestion
			. += "It's full of patched up holes!"
```
grammar

frawhst reviewed (commented):
few minor things
